### DISCUSSION TITLE
Does --force-cpu actually work

### DISCUSSION THREAD

**Discussion by econundrum** [d8ahazard/sd_dreambooth_extension#951] [General]
Since they took away the CPU training option has anyone actually managed to get it to work using --force-cpu ?

I still get this error as if it's trying to allocate vRAM

 return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None, non_blocking)
RuntimeError: CUDA out of memory. Tried to allocate 58.00 MiB (GPU 0; 4.00 GiB total capacity; 3.48 GiB already allocated; 0 bytes free; 3.53 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
Traceback (most recent call last):

this is my dreambooth.bat

echo off

set PYTHON=
set GIT=
set VENV_DIR=
set COMMANDLINE_ARGS= --force-cpu
:: Use the below argument if getting OOM extracting checkpoints
:: set COMMANDLINE_ARGS=--ckptfix
set "REQS_FILE=.\extensions\sd_dreambooth_extension\requirements.txt"
:: Uncomment below to skip trying to install automatically on launch.
:: set "DREAMBOOTH_SKIP_INSTALL=True"
:: Use this to launch with accelerate (Run 'accelerate config' first, launch once without to install dependencies)
:: set ACCELERATE="True"

call webui.bat

**Comment by econundrum** [comment]
Deeply frustrating CPU training used to be really easy to do now I'm playing with command line switches that don't appear to work.

**Comment by Arilziem** [comment]
Never tried it, but maybe you need to set both `--force-cpu` and `--ckptfix`? The first should imply the second but I'm not sure if it's implemented that way. If you get that error while creating the initial dreambooth model, try adding the ckptfix flag (it forces loading the model into RAM instead of VRAM during model extraction).
